### PR TITLE
remove transformer packages in favor of `@itwin/transformer` monorepo

### PR DESCRIPTION
**Pull request by MichaelBelousov**
- The transformer has a new home at [`@itwin/transformer`](https://github.com/iTwin/transformer/)
- The test-app lives there too now, so removing it
- Also adds transformer docs pipeline integration (thanks @skirby1996)
- fixed doc references, kept learning docs in here for now like other externalized repos are doing

### CODE REVIEW

File: docs/changehistory/ChangeLogs.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 - [core-backend](../reference/core-backend/changelog)
 
-- [core-transformer](../reference/core-transformer/changelog)
+- [transformer](../reference/transformer/changelog)
```
Comment (aruniverse) [inline docs/changehistory/ChangeLogs.md:5]: NextVersion.md?

**Comment by pmconne** [comment]
I'm curious if the package name change from `core-transformer` to `transformer` was discussed and is set in stone now.
I ask because it's a breaking change for anyone currently consuming the package and if a name change is desired/needed, was something more descriptive considered like `imodel-transformer`?

**Comment by MichaelBelousov** [comment]
> I'm curious if the package name change from `core-transformer` to `transformer` was discussed and is set in stone now. I ask because it's a breaking change for anyone currently consuming the package and if a name change is desired/needed, was something more descriptive considered like `imodel-transformer`?

It was discussed, but it's (kind of?) too late now, @itwin/transformer is already published (but completely unused so theoretically undo-able via imjs-admin). We went with a new name because if we kept the old one, due to internal API usage, separate patch timelines for core and this package could result in a situation where depending upon the same patch version of `@itwin/core-backend` and `@itwin/core-transformer` could be incompatible which we thought was confusing. Now that you mention it, don't the now version-unlocked presentation and appui libraries have the same problem and they definitely of course kept their names (they didn't have core in their names to drop, however)

CC: @calebmshafer

**Comment by MichaelBelousov** [comment]
renaming the new package to `@itwin/imodel-transformer`

**Pull request closed without merging**